internal/pkg/agent/install/install.go

```diff
@@ -125,6 +125,14 @@ func Install(cfgFile, topPath string, unprivileged bool, log *logp.Logger, pt *p
 		return utils.FileOwner{}, err
 	}
 
+	// create the install marker
+	// do this after the file copy but before installing service files, etc
+	// That way if the service install process fails, a user can still run `elastic-agent uninstall`
+	// to clean up.
+	if err := CreateInstallMarker(topPath, ownership); err != nil {
```

Comment:
This is still happening after the `copyFiles` call, it should be immediately after the `err = os.MkdirAll(filepath.Dir(topPath), 0755)` earlier in this function.

Reply:
So, I chose to not put it there, since putting the `.installed` file before anything has been installed seemed counter-intuitive, and if the copy fails we might not be able to run `elastic-agent` at all. Unless there's something I'm not factoring in?

Reply:
You can't run `elastic-agent uninstall` unless the `.installed` marker exists, so if you put it after the copy you can't remove the copied files from disk with the `elastic-agent uninstall` command.